galera/src/replicator_str.cpp

```diff
@@ -668,10 +672,15 @@ ReplicatorSMM::send_state_request (const StateRequest* const req)
         }
         else
         {
-            // connection is being closed, send failure is expected
-            st_.mark_safe();
+            // connection is being closed, send failure is expected.
+            if (unsafe)
```

Comment:
shouldn't this condition be

if (!unsafe)... otherwise it will try to re-set state = unsafe which as you notified will raise an assertion.